src/tests/DevToys.Tests/Providers/Tools/JwtEncoderTests.cs

```diff
@@ -0,0 +1,521 @@
+                TokenAlgorithm = JwtAlgorithm.ES256,
+                Payload = await TestDataProvider.GetFileContent("Jwt.BasicPayload.json"),
+                PrivateKey = privateKey
+            };
+            var jwtDecoder = new JwtEncoder();
+            TokenResult result = jwtDecoder.GenerateToken(encoderParameters, tokenParameters, DecodingErrorCallBack);
+
+            Assert.IsNotNull(result);
+            Assert.IsNull(_validationResult);
+            Assert.IsNotNull(result.Token);
+        }
+
+        [TestMethod]
+        public async Task JwtEncoder_Generate_Complex_ES_Token()
+        {
+            var encoderParameters = new EncoderParameters()
+            {
+                HasAudience = true,
+                HasExpiration = true,
+                HasIssuer = true,
+                HasDefaultTime = true
+            };
+            var tokenParameters = new TokenParameters()
+            {
+                TokenAlgorithm = JwtAlgorithm.ES256,
+                Payload = await TestDataProvider.GetFileContent("Jwt.BasicPayload.json"),
+                PrivateKey = await TestDataProvider.GetFileContent("Jwt.ES.PrivateKey.txt"),
+                ValidIssuers = new HashSet<string> { "devtoys" },
+                ValidAudiences = new HashSet<string> { "devtoys" },
+                ExpirationYear = 1,
+                ExpirationMonth = 1,
+                ExpirationDay = 1,
+                ExpirationHour = 0,
+                ExpirationMinute = 0
+            };
+
+            var jwtDecoder = new JwtEncoder();
+            TokenResult result = jwtDecoder.GenerateToken(encoderParameters, tokenParameters, DecodingErrorCallBack);
+
+            Assert.IsNotNull(result);
+            Assert.IsNull(_validationResult);
+            Assert.IsNotNull(result.Token);
+        }
+
+        [TestMethod]
+        public async Task JwtEncoder_Generate_Basic_ES_Token_With_Public_Key_Should_Faild()
+        {
+            var encoderParameters = new EncoderParameters();
+            var tokenParameters = new TokenParameters()
+            {
+                TokenAlgorithm = JwtAlgorithm.ES256,
+                Payload = await TestDataProvider.GetFileContent("Jwt.BasicPayload.json"),
+                PrivateKey = await TestDataProvider.GetFileContent("Jwt.ES.PublicKey.txt")
+            };
+            var jwtDecoder = new JwtEncoder();
+            TokenResult result = jwtDecoder.GenerateToken(encoderParameters, tokenParameters, DecodingErrorCallBack);
+
+            Assert.IsNull(result);
+            Assert.IsNotNull(_validationResult);
+            Assert.AreEqual(_validationResult.Severity, InfoBarSeverity.Error);
+        }
+
+        #endregion
+
+        internal static void DecodingErrorCallBack(TokenResultErrorEventArgs e)
+             => _validationResult = new InfoBarData(e.Severity, e.Message);
+
+        [TestCleanup]
+        public void TestCleanup()
+        {
+            _validationResult = null;
+        }
+    }
+}
```

Comment:
Kudo for adding 60+ tests :D This is great! I see 8 tests failing on my machine though 🤔

![image](https://user-images.githubusercontent.com/3747805/187049316-2b8d02fd-407b-4b37-a8fa-6613213b9ac1.png)

![image](https://user-images.githubusercontent.com/3747805/187049319-57d01c23-887f-4457-af79-7afa33876595.png)

Do they pass on your machine? If yes, perhaps it's the sign of a race condition.

Reply:
Hum these tests Passed in devops too. Can you get the errors?

Reply:
Hum, that's the problem with the expires date and time zone. I'm checking.